jupyter-notebooks/scipy-2022-workshop/3_Compute_NDWI.ipynb

```diff
@@ -0,0 +1,267 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "image_file = \"/content/20170831_172754_101c_3B_AnalyticMS.tif\"\n",
+    "satdat = rasterio.open(image_file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calculate NDWI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we're looking at flooding in this example, it would be useful to identify areas of water v.s. areas of land.\n",
+    "\n",
+    "We can measure the normalized difference water index (NDWI), which is related to plant water content and helps identify areas of water. \n",
+    "\n",
+    "NDWI can be calculated as: NDWI = (green - NIR) / (green + NIR)\n",
+    "\n",
+    "You can find more information on NDWI here: https://en.wikipedia.org/wiki/Normalized_difference_water_index"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, we need to allow division by zero when we do our NDWI calculation. We do this because our image may have masked or clipped values that all have values of zero, where the sum of those values in the denominator can also be zero. We want to make sure that our code doesn't break if this happens, so we want to allow numpy to divide by zero."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Allow division by zero\n",
+    "np.seterr(divide='ignore', invalid='ignore')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Then, we need to extract the bands we will need for our NDWI calculation. In this case, we want to extract the green and near-infrared bands."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "blue, green, red, nir = satdat.read()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Calculate NDWI\n",
+    "ndwi = (green.astype(float) - nir.astype(float)) / (green + nir)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We now have an array of NDWI values we can use for further analysis!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing NDWI"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now plot our NDWI to visualize it, similar to what we did with our original image."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(0)\n",
+    "plt.imshow(ndwi)\n",
```

Comment:
I'd recommend setting the colour map to "blues", as is done in the `coastline_analysis` notebook. It really helps the reader visualize where the water resides in the maps.
```suggestion
    "fig = plt.imshow(ndwi)\n",
    "fig.set_cmap('Blues')\n",
```